src/roswire/common/source.py

```diff
@@ -0,0 +1,139 @@
+# -*- coding: utf-8 -*-
+__all__ = (
+    "extract_sources_from_cmake",
+    "NodeSourceInfo",
+    "PackageSourceExtractor",
+    "SourceLanguage",)
+
+import abc
+import enum
+import re
+import typing as t
+from pathlib import Path
+from typing import Any, Iterable  # noqa: F401, E501 # Needed for tuple_from_iterable and argparse
+
+import attr
+
+from .cmake import (
+    argparse as cmake_argparse,
+    ParserContext as CMakeParserContext,
+)
+from ..util import tuple_from_iterable
+
+if t.TYPE_CHECKING:
+    from .. import AppInstance
+
+
+class SourceLanguage(enum.Enum):
+    CXX = "cxx"
+    PYTHON = "python"
+
+
+@attr.s(frozen=True, auto_attribs=True, slots=True)
+class NodeSourceInfo:
+    node_name: str
+    language: SourceLanguage
+    sources: t.Collection[str] = attr.ib(converter=tuple_from_iterable)
+
+    def to_dict(self) -> t.Dict[str, t.Any]:
+        return {"name": self.node_name,
+                "language": self.language.value,
+                "sources": list(self.sources),
+                }
+
+    @classmethod
+    def from_dict(cls, info: t.Dict[str, t.Any]) -> "NodeSourceInfo":
+        return NodeSourceInfo(
+            info["name"],
+            SourceLanguage(info["language"]),
+            info["sources"]
+        )
+
+
+def extract_sources_from_cmake(
+        file_contents: str
```

Comment:
nitpick: indentation should be four spaces here.

Reply:
Yeah. Its something I always have to remember to undo when I use autoformatting because I can't work out how to get it to format to only 4 spaces in argument lists.

Reply:
Done